library/src/main/java/com/novoda/downloadmanager/NotificationConfig.java

```diff
@@ -0,0 +1,53 @@
+package com.novoda.downloadmanager;
+
+import android.app.Notification;
+import android.app.NotificationChannel;
+import android.content.Context;
+import android.os.Build;
+import android.support.v4.app.NotificationCompat;
+
+public class NotificationConfig<T> implements NotificationChannelCreator, NotificationCreator<T> {
+
+    private final Context applicationContext;
+    private final String channelId;
+    private final String userFacingChannelDescription;
+    private final NotificationCustomizer<T> notificationCustomizer;
+    private final int importance;
+
+    public NotificationConfig(Context context, String channelId, String userFacingChannelDescription, NotificationCustomizer<T> customizer, int importance) {
```

Comment:
Do you know if its possible to annotate the `int importance` param so that we know the valid range of values?

Reply:
Yes we can do that!